src/models/model.h

```diff
@@ -31,7 +31,11 @@ struct State {
   virtual RoamingArray<float> Run(int current_length, RoamingArray<int32_t> next_tokens, RoamingArray<int32_t> next_indices = {}) = 0;
   virtual const CapturedGraphInfo* GetCapturedGraphInfo() const { return nullptr; }
 
-  OrtValue* GetOutput(const char* name);
+  OrtValue* GetInput(const char* name);
+
+  virtual OrtValue* GetOutput(const char* name);
```

Comment:
@kunal-vaishnavi Just FYI since in your whisper changes you added 'GetSpecialOutput' that is called when 'GetOutput' has no match, this should work for your case also.